### PR TITLE
Rename optimization loops

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

The internal OptimizerLoop class does no longer represent a real loop after #16539. We rename `OptimizerLoop` and `ManualLoop` to just `AutomaticOptimization` and `ManualOptimization`, which is more representative of what they do.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @justusschock @awaelchli @borda @carmocca

### CODE REVIEW

File: src/lightning/pytorch/loops/optimization/manual.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def run(self, kwargs: OrderedDict) -> _OUTPUTS_TYPE:
 
     def on_run_start(self) -> None:
         # inject logic around the optimizer step
-        for lightning_optimizer in self.trainer.strategy._lightning_optimizers:
+        for i, lightning_optimizer in self.trainer.strategy._lightning_optimizers.items():
```
Comment (awaelchli) [inline src/lightning/pytorch/loops/optimization/manual.py:95]:
```suggestion
        for lightning_optimizer in self.trainer.strategy._lightning_optimizers:
```

File: src/lightning/pytorch/loops/optimization/manual.py [hunk 2]
```diff
@@ -119,7 +119,7 @@ def on_run_end(self) -> _OUTPUTS_TYPE:
         """Returns the result of this loop, i.e., the post-processed outputs from the training step."""
         output, self._output = self._output, {}  # free memory
         # reset logic around the optimizer step
-        for lightning_optimizer in self.trainer.strategy._lightning_optimizers:
+        for i, lightning_optimizer in self.trainer.strategy._lightning_optimizers.items():
```
Comment (awaelchli) [inline src/lightning/pytorch/loops/optimization/manual.py:122]:
```suggestion
        for lightning_optimizer in self.trainer.strategy._lightning_optimizers:
```

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: src/lightning/pytorch/loops/epoch/training_epoch_loop.py [hunk 3]
```diff
@@ -68,7 +68,7 @@ def __init__(self, min_steps: Optional[int] = None, max_steps: int = -1) -> None
         self.batch_progress = BatchProgress()
         self.scheduler_progress = SchedulerProgress()
 
-        self.optimizer_loop = _OptimizerLoop()
+        self.optimizer_loop = _AutomaticOptimization()
```
Comment (justusschock) [inline src/lightning/pytorch/loops/epoch/training_epoch_loop.py:71]: nit: can we also rename these variables then?

**Review by justusschock — APPROVED** [review]

File: src/lightning/pytorch/loops/epoch/training_epoch_loop.py (hunk 3 above)
Comment (Borda) [reply src/lightning/pytorch/loops/epoch/training_epoch_loop.py:71]: they are public :/

Comment (awaelchli) [reply src/lightning/pytorch/loops/epoch/training_epoch_loop.py]: Thanks for spotting this. It was incorrectly undone when I was resolving merge conflicts.
